apps/api/src/domain/models/vocabulary-list/commands/register-vocabulary-list-filter-property/register-vocabulary-list-filter-property.command-handler.ts

```diff
@@ -0,0 +1,55 @@
+import { CommandHandler } from '@coscrad/commands';
+import { Valid } from '../../../../../domain/domainModelValidators/Valid';
+import { DeluxeInMemoryStore } from '../../../../../domain/types/DeluxeInMemoryStore';
+import { InMemorySnapshot } from '../../../../../domain/types/ResourceType';
+import { InternalError } from '../../../../../lib/errors/InternalError';
+import { ResultOrError } from '../../../../../types/ResultOrError';
+import { BaseUpdateCommandHandler } from '../../../shared/command-handlers/base-update-command-handler';
+import { BaseEvent } from '../../../shared/events/base-event.entity';
+import { VocabularyList } from '../../entities/vocabulary-list.entity';
+import { DropboxOrCheckbox } from '../../types/dropbox-or-checkbox';
+import {
+    FilterPropertyType,
+    RegisterVocabularyListFilterProperty,
+} from './register-vocabulary-list-filter-property.command';
+import { VocabularyListFilterPropertyRegistered } from './vocabulary-list-filter-property-registered';
+
+@CommandHandler(RegisterVocabularyListFilterProperty)
+export class RegisterVocabularyListFilterPropertyCommandHandler extends BaseUpdateCommandHandler<VocabularyList> {
+    protected async fetchRequiredExternalState(
+        _: RegisterVocabularyListFilterProperty
+    ): Promise<InMemorySnapshot> {
+        return Promise.resolve(new DeluxeInMemoryStore({}).fetchFullSnapshotInLegacyFormat());
+    }
+
+    protected validateExternalState(
+        _state: InMemorySnapshot,
+        _instance: VocabularyList
+    ): InternalError | Valid {
+        return Valid;
+    }
+
+    protected actOnInstance(
+        instance: VocabularyList,
+        { name, type, allowedValuesAndLabels }: RegisterVocabularyListFilterProperty
+    ): ResultOrError<VocabularyList> {
+        /**
+         * This is necessary until a migration \ event versioning occurs so we can
```

Comment:
[Here](https://www.pivotaltracker.com/story/show/186469235) is the story for that.